tests/dataconverter/test_validation.py

```diff
@@ -1694,27 +1711,64 @@ def listify_template(data_dict: Template):
             id="reserved-prefix",
         ),
         pytest.param(
-            alter_dict(
+            compress_paths_in_dict(
                 TEMPLATE,
-                "/ENTRY[my_entry]/NXODD_name[nxodd_name]/float_value",
+                [
+                    "/ENTRY[my_entry]/NXODD_name[nxodd_name]/float_value"
+                    "/ENTRY[my_entry]/NXODD_name[nxodd_name]/number_value",
+                    "/ENTRY[my_entry]/NXODD_name[nxodd_name]/bool_value",
+                    "/ENTRY[my_entry]/NXODD_name[nxodd_name]/int_value",
+                    "/ENTRY[my_entry]/NXODD_name[nxodd_name]/posint_value",
+                    "/ENTRY[my_entry]/NXODD_name[nxodd_name]/char_value",
+                    "/ENTRY[my_entry]/NXODD_name[nxodd_name]/date_value",
+                    "/ENTRY[my_entry]/NXODD_name[nxodd_name]/type",
+                ],
+            ),
+            [
+                "Compression for /ENTRY[my_entry]/NXODD_name[nxodd_name]/bool_value = {'compress': True, 'strength': 3} "
+                "should not be used for boolean values.",
+                "Compression for /ENTRY[my_entry]/NXODD_name[nxodd_name]/char_value = {'compress': 'just chars', 'strength': 3} "
+                "should not be used for string values.",
+                "Compression for /ENTRY[my_entry]/NXODD_name[nxodd_name]/date_value = {'compress': '2022-01-22T12:14:12.05018+00:00', 'strength': 3} "
+                "should not be used for string values.",
+                "Compression for /ENTRY[my_entry]/NXODD_name[nxodd_name]/type = {'compress': '2nd type', 'strength': 3} "
+                "should not be used for enumerated concepts.",
+            ],
+            id="appdef-compressed",
+        ),
+        pytest.param(
+            alter_dict(
+                alter_dict(
+                    TEMPLATE,
+                    "/ENTRY[my_entry]/NXODD_name[nxodd_name]/float_value",
+                    {"compress": np.int64(2.0), "strength": 1},
+                ),
+                "/ENTRY[my_entry]/NXODD_name[nxodd_name]/int_value",
                 {"compress": np.float32(2.0), "strength": 1},
             ),
-            [],
-            id="appdef-compressed-payload",
+            [
+                "The value at /ENTRY[my_entry]/NXODD_name[nxodd_name]/int_value "
+                "should be one of the following Python types: "
+                "(<class 'int'>, <class 'numpy.integer'>), as defined in the "
+                "NXDL as NX_INT."
+            ],
+            id="appdef-compressed-wrong-type",
         ),
         pytest.param(
             alter_dict(
```

Comment:
Adding one test probing strength = 0 is useful to test the aforementioned warning branch in cases of expectable low compression ratio